src/init.cc

```diff
@@ -422,21 +425,72 @@ NCCL_PARAM(DmaBufEnable, "DMABUF_ENABLE", 0);
 // Detect DMA-BUF support
 static ncclResult_t dmaBufSupported(struct ncclComm* comm) {
   if (ncclParamDmaBufEnable() == 0 || comm->ncclNet->regMrDmaBuf == NULL || rocmLibraryInit() != ncclSuccess) return ncclInternalError;
-#if CUDA_VERSION >= 11070
-  int flag = 0;
-  CUdevice dev;
-  int cudaDriverVersion;
-  CUDACHECK(cudaDriverGetVersion(&cudaDriverVersion));
-  if (CUPFN(cuDeviceGet) == NULL || cudaDriverVersion < 11070) return ncclInternalError;
-  CUCHECK(cuDeviceGet(&dev, comm->cudaDev));
-  // Query device to see if DMA-BUF support is available
-  (void) CUPFN(cuDeviceGetAttribute(&flag, CU_DEVICE_ATTRIBUTE_DMA_BUF_SUPPORTED, dev));
-  if (flag == 0) return ncclInternalError;
-  INFO(NCCL_INIT, "DMA-BUF is available on GPU device %d", comm->cudaDev);
-  return ncclSuccess;
-#else
-  return pfn_hsa_amd_portable_export_dmabuf != NULL ? ncclSuccess : ncclInternalError;
-#endif
+  #if CUDA_VERSION >= 11070
+    int flag = 0;
+    CUdevice dev;
+    int cudaDriverVersion;
+    CUDACHECK(cudaDriverGetVersion(&cudaDriverVersion));
+    if (CUPFN(cuDeviceGet) == NULL || cudaDriverVersion < 11070) return ncclInternalError;
+    CUCHECK(cuDeviceGet(&dev, comm->cudaDev));
+    // Query device to see if DMA-BUF support is available
+    (void) CUPFN(cuDeviceGetAttribute(&flag, CU_DEVICE_ATTRIBUTE_DMA_BUF_SUPPORTED, dev));
+    if (flag == 0) return ncclInternalError;
+    INFO(NCCL_INIT, "DMA-BUF is available on GPU device %d", comm->cudaDev);
+    return ncclSuccess;
+  #else
+    //Rocm support check
+    hsa_status_t status = hsa_init();
+    int rocmSupport, kernelSupport = 0;
+    bool value = false;
+
+    //look for HSA_AMD_SYSTEM_INFO_DMABUF_SUPPORTED in hsa.h and check if it is supported
+    status = hsa_system_get_info((hsa_system_info_t) 0x204, &value);
+
+    if (status == HSA_STATUS_SUCCESS) {
+        if (value) rocmSupport = 1;
+        else INFO(NCCL_ALL, "Rocm: DMA_BUF Support Failed");
+    }
+    else if (status == HSA_STATUS_ERROR_NOT_INITIALIZED) INFO(NCCL_ALL, "DMA_BUF ERROR: The HSA runtime has not been initialized.");
+    else if (status == HSA_STATUS_ERROR_INVALID_ARGUMENT) INFO(NCCL_ALL, "DMA_BUF ERROR: Invalid argument. HSA attribute is an invalid system attribute, or value is NULL.");
```

Comment:
We can simplify this error message so that users know that the current version of ROCm they have doesn't support DMABUF.
if (status != HSA_STATUS_SUCCESS || !value)
{
  INFO(NCCL_ALL, "Current version of ROCm does not support dmabuf feature.")
}